modules/common/development/cuda.nix

```diff
@@ -0,0 +1,24 @@
+# Copyright 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{ lib, config, ... }:
+let
+  cfg = config.ghaf.development.cuda;
+  inherit (lib) mkEnableOption mkIf;
+in
+{
+  options.ghaf.development.cuda = {
+    enable = mkEnableOption "CUDA Support";
+  };
+
+  config = mkIf cfg.enable {
+    #Enabling CUDA on any supported system requires below settings. 
+    nixpkgs.config.allowUnfree = lib.mkForce true;
```

Comment:
Did we not pass in a high-level nixpkgs instance or is this false memory?
In that case nixpkgs.config wouldn't work.

Reply:
But I think if there is no error, than the answer is no.